swift/ql/lib/codeql/swift/controlflow/internal/Completion.qll

```diff
@@ -122,9 +122,9 @@ private predicate inBooleanContext(ControlFlowElement n) {
 private predicate astInBooleanContext(AstNode n) {
   n = any(ConditionElement condElem).getBoolean().getFullyUnresolved()
   or
-  n = any(ConditionElement condElem).getAvailability().getFullyUnresolved()
+  n = any(ConditionElement condElem).getAvailability()
```

Comment:
An open question:

Looking through the changes to CFG construction, we've had bugs in the past because someone (most likely me 😅) forgot to call recursively on the fully unresolved / fully converted sub-element which led to inconsistent CFGs. And in those cases it was pretty easy to see syntactically that:
- We were calling `getFullyUnresolved` (or `getFullyConverted`) in 99% of the cases, and
- 1% of the cases were missing a call to `getFullyUnresolved` (or `getFullyConverted`)

And so that 1% was most likely a bug. But now some places will need the to call `getFullyUnresolved` (when it's available), and other places won't need to do that (because there's no such method on the class).

Do you think this is a problem?

Reply:
I backtracked on this on a later commit. As I understood that "hideability" travels both directions of a hierarchy, `Element` ended up hideable no matter what, so in the end I kept the `resolve`/`getFullyUnresolved` predicates on `Element`, and undid all these changes. I agree that that was a possible source of bugs.

Sorry for the mix up.